app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class BaseController < ApplicationController
+      skip_before_action :verify_authenticity_token
+
+      private
+
+      def require_authentication!
+        user = User.find_by(authentication_token: request.headers["X-Api-Key"])
```

Comment:
I suggest using `Authorization: Bearer <token>`. It will also make some log services filter it out automatically.